src/core/qlist.h

```diff
@@ -19,38 +20,70 @@ class QList {
   enum Where { TAIL, HEAD };
 
   // Provides wrapper around the references to the listpack entries.
-  struct Entry {
-    Entry(const char* value, size_t length) : value{value}, length{length} {
+  class Entry {
+    std::variant<std::string_view, int64_t> value_;
+
+   public:
+    Entry(const char* value, size_t length) : value_{std::string_view(value, length)} {
     }
-    Entry(long long longval) : value{nullptr}, longval{longval} {
+
+    explicit Entry(int64_t longval) : value_{longval} {
     }
 
-    // Assumes value is not null.
+    // Assumes value is not int64.
     std::string_view view() const {
-      return {value, length};
+      return std::get<std::string_view>(value_);
     }
 
-    const char* value;
-    union {
-      size_t length;
-      long long longval;
-    };
+    int64_t ival() const {
+      return std::get<int64_t>(value_);
+    }
+
+    bool operator==(std::string_view sv) const;
+
+    std::string to_string() const {
+      if (std::holds_alternative<int64_t>(value_)) {
+        return std::to_string(std::get<int64_t>(value_));
+      }
+      return std::string(view());
+    }
+  };
+
+  class Iterator {
+   public:
+    Entry Get() const;
+
+    // Returns false if no more entries.
+    bool Next();
```

Comment:
HasNext was a better name

Reply:
it also moves the iterator. HasNext sounds like read-only. I am just mirroring the quicklist API at this point